### PR TITLE
Add hostfactory integration tests and improve UX

### PR DESCRIPTION
**Pull request by sigalsax**
### What does this PR do?
This commit adds integration tests for the hostfactory vanilla flow as well as improve UX for the hostfactory command. Previously the user had to pass in the account and command kind with the CLI command (cucumber:host_factory:somepolicy/my_host_factory) and now they can just do so by providing the name of the hostfactory path (somepolicy/my_host_factory)

### What ticket does this PR close?
Resolves - Jira

#### Change log
- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Added
+- The `hostfactory` methods `create token` are now available in CLI and SDK to create a hostfactory token
```
Comment (orenbm) [inline CHANGELOG.md:10]:
```suggestion
- The `hostfactory` method `create token` is now available in CLI and SDK to create a hostfactory token 
```

File: test/test_config/hostfactory_policy.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  id: hostfactory_policy
+
+  body:
+    - !layer some_vms
+
+    - !host-factory
+      id: some_host_factory
+      annotations:
+        description: Factory to create new vms
+      layers: [ !layer some_vms ]
```
Comment (orenbm) [inline test/test_config/hostfactory_policy.yml:11]: missing new line

File: test/test_config/hostfactory_policy.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+- !policy
+  id: hostfactory_policy
+
+  body:
+    - !layer some_vms
```
Comment (orenbm) [inline test/test_config/hostfactory_policy.yml:5]: can we use `!group` instead? i think that `layer` will be deprecated sometime in the future

Comment (orenbm) [reply test/test_config/hostfactory_policy.yml:5]: can `host-factory` work with `group`?

File: test/test_integration_hostfactory.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    # *************** TESTS ***************
+
+    @integration_test()
+    def test_hostfactory_vanilla_returns_correct_response(self):
+        output = self.invoke_cli(self.cli_auth_params,
+                                 ['hostfactory', 'create', 'token', '-i', 'hostfactory_policy/some_host_factory'])
+
+        self.assertIn('[\n    {\n        "cidr": [],\n'
+                      f'        "expiration": "{(datetime.utcnow().replace(microsecond=0) + timedelta(hours=1)).isoformat()}Z",\n'
+                      '        "token":', output)
```
Comment (orenbm) [inline test/test_integration_hostfactory.py:39]: can we make this look better?

File: test/test_config/hostfactory_policy.yml (hunk 3 above)
Comment (sigalsax) [reply test/test_config/hostfactory_policy.yml:5]: yes, just checked

File: test/test_integration_hostfactory.py (hunk 4 above)
Comment (sigalsax) [reply test/test_integration_hostfactory.py:39]: the formatting is important

**Review by eladkug — COMMENTED** [review]
LGTM for testing.
We will need to add more tests / unit tests when we will start to work with our flags.

File: test/test_config/hostfactory_policy.yml (hunk 3 above)
Comment (sigalsax) [reply test/test_config/hostfactory_policy.yml:5]: checked with group and get `422 Client Error: Unprocessable Entity for url: https://localhost/policies/cucumber/policy/root. The policy was empty or its contents were not in valid YAML.`

Comment (orenbm) [reply test/test_config/hostfactory_policy.yml:5]: that's odd. can you share the policy you tried to load?

Comment (sigalsax) [reply test/test_config/hostfactory_policy.yml:5]:
```
- !policy
  id: hostfactory_policy

  body:
    - !group some_vms

    - !host-factory
      id: some_host_factory
      annotations:
        description: Factory to create new vms
      groups: [ !group some_vms ]

```

**Review by orenbm — APPROVED** [review]

**Pull request merged**